common/walker/src/report.rs

```diff
@@ -123,9 +126,97 @@ impl SplitScannerError for Result<Report, ScannerError> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
+pub enum ReportType {
+    SBOM(String),
+    VEX(String),
+}
 /// Handle the report
-pub async fn handle_report(report: Report) -> anyhow::Result<()> {
-    // FIXME: this is a very simplistic version of handling the error
-    log::info!("Import report: {report:#?}");
+pub async fn handle_report(report: Report, report_type: ReportType) -> anyhow::Result<()> {
+    let path = env::var_os("REPORT_PATH").unwrap_or_else(|| OsString::from(DEFAULT_REPORT_OUTPUT_PATH));
+    let path = path.to_str().unwrap_or(DEFAULT_REPORT_OUTPUT_PATH).to_string();
+
+    let report_type_string = match report_type {
+        ReportType::SBOM(sbom) => sbom,
+        ReportType::VEX(vex) => vex,
+    };
+
+    let template_content = include_str!("../templates/report.html");
+    let mut tera = Tera::default();
+    tera.add_raw_template("report.html", template_content)?;
+
+    let mut context = Context::new();
+    let current_time = OffsetDateTime::now_utc();
+    context.insert("report", &report);
+    context.insert("type", &report_type_string);
+    context.insert("current_time", &current_time);
+
+    let format = format_description!("[year]-[month]-[day].[hour].[minute].[second]");
+    let formatted_time = current_time.format(&format)?;
+    match tera.render("report.html", &context) {
+        Ok(rendered_html) => {
+            let out_put_html_path = format!("{path}/{report_type_string}/html/report-{formatted_time}.html");
+            if let Some(parent) = std::path::Path::new(&out_put_html_path).parent() {
+                fs::create_dir_all(parent)?;
+            }
+            let mut file = match File::create(out_put_html_path.clone()) {
+                Ok(file) => file,
+                Err(e) => {
+                    log::warn!(" The {} created failed. {:?}", out_put_html_path.clone(), e);
+                    return Err(e.into());
+                }
+            };
+
+            let _ = writeln!(file, "{}", rendered_html);
+            let _ = file.sync_all();
+            log::info!("Successfully generated the report file.");
```

Comment:
If the previous file operations fail, that's not the case. We should log information that might be potentially wrong, which makes it hard to trust the application and search for possible errors.